NUnitLite-1.0.0/src/framework/Internal/Results/TestResult.cs

```diff
@@ -412,20 +413,27 @@ public void RecordException(Exception ex)
             if (ex is NUnitException)
                 ex = ex.InnerException;
 
-            if (ex is System.Threading.ThreadAbortException)
+            if (ex is System.Threading.ThreadAbortException) {
               SetResult(ResultState.Cancelled, "Test cancelled by user", ex.StackTrace);
-            else if (ex is AssertionException)
+            } else if (ex is AssertionException) {
               SetResult(ResultState.Failure, ex.Message, StackFilter.Filter(ex.StackTrace));
-            else if (ex is IgnoreException)
+            } else if (ex is IgnoreException) {
               SetResult(ResultState.Ignored, ex.Message, StackFilter.Filter(ex.StackTrace));
-            else if (ex is InconclusiveException)
+            } else if (ex is InconclusiveException) {
               SetResult(ResultState.Inconclusive, ex.Message, StackFilter.Filter(ex.StackTrace));
-            else if (ex is SuccessException)
+            } else if (ex is SuccessException) {
               SetResult(ResultState.Success, ex.Message, StackFilter.Filter(ex.StackTrace));
-            else
-                SetResult(ResultState.Error,
+            } else {
+              MethodInfo write = null;
+              if (Environment.GetEnvironmentVariable ("MONO_TEST_TELEMETRY") != null)
```

Comment:
What is the purpose of this? If you want to test the telemetry it's better to write non-unit test

Reply:
The purpose of this is to serialize unhandled exceptions to disk. I assumed this was not a default desired behavior, so I placed the functionality behind an environment variable. This isn't a unit test.

Reply:
This is being used on CI to send crashes to sentry

Reply:
I am not sure just how this will be usefull when there are many failures. Shouldn't there be a way to tweak the random file name?